### ISSUE TITLE
Warning is not showing up?

### ISSUE THREAD

**Issue by why-jay** [chcokr/remark-lint-sentence-newline#2]
I've seen cases where the [file.warn](https://github.com/chcokr/mdast-lint-sentence-newline/blob/master/index.js#L22) is reached but at the end of the linting I don't see any yellow warnings from this rule on the CLI.

@wooorm Could you do me a favor and see if there is more than what I did in the code in order to render a warning on the CLI?

**Comment by wooorm** [comment]
I’m not sure what use case you were referring to. I’m assuming regarding to [mdast-lint-sentence-newline](https://github.com/chcokr/mdast-lint-sentence-newline)? I ran your example and it was ignored because HTML needs an extra black line after it:

The following is just two HTML blocks:

``` md
<!-- Invalid -->
Hello, world. This sentence should be on a second line.

<!-- Valid -->
Hello, world.
This sentence should be on a second line.
```

But this is HTML comments and text:

``` md
<!-- Invalid -->

Hello, world. This sentence should be on a second line.

<!-- Valid -->

Hello, world.
This sentence should be on a second line.
```

The last one is warned about about:

![screen shot 2015-09-12 at 9 13 26 a m](https://cloud.githubusercontent.com/assets/944406/9830438/8f54a3be-592e-11e5-9d50-dad4b97db344.png)

**Comment by why-jay** [comment]
@wooorm Sorry for not being clearer - yes, I am talking about my custom rule. And, ah, I didn't know about the HTML comments thing, thanks. Just pushed a commit that fixes the example. 

But there's another test case where what I mentioned above is still happening - let me come back with a test case to show you.

**Comment by why-jay** [comment]
Fuck, I'm too tired - I'll ping you tomorrow. Thanks for your help.

**Comment by wooorm** [comment]
Sure!

**Comment by why-jay** [comment]
@wooorm I'm writing tests - can I get some more help? According to the `mdast-lint` API doc, I'm supposed to use `file.messages` inside the callback to `#process()`. But on this line in my test suite, I keep getting `undefined` for `file.messages`:

https://github.com/chcokr/mdast-lint-sentence-newline/blob/master/test/test.js#L26

**Comment by wooorm** [comment]
`err, res, file` > `err, file, res`. Did you get this from an example?

**Comment by why-jay** [comment]
Ah, thanks! Yes, from here: https://github.com/wooorm/mdast-lint/blob/master/doc/api.md

Looks like it needs to be fixed over there!

**Comment by wooorm** [comment]
on it

**Comment by why-jay** [comment]
@wooorm Thanks, you the man.

Alright, so I just added my mysterious test case. It's actually not mysterious any more - I know why it's breaking. When there's a header, `unist-util-visit` doesn't seem to visit the section's child `text` nodes properly. Shouldn't it visit all `text` nodes, even ones in children, when I do `visit(ast, 'text')`? 

https://github.com/chcokr/mdast-lint-sentence-newline/blob/master/test/header-body/file.md

**Comment by wooorm** [comment]
Shouldn’t the `done();` be outside `visit`? Maybe that’s it?

**Comment by wooorm** [comment]
Yup, changing that fixed your tests.

**Comment by why-jay** [comment]
Ah, silly me! Closed via 4c302fdaf3feac625876d31ce5395320dd961204. Thank you so much for all your work - `mdast` is very well maintained and I hope it gets even more attention and adoption. Is there any way to donate to your open-source efforts?
